contracts/BaseWrapper.sol

```diff
@@ -253,7 +253,7 @@ abstract contract BaseWrapper {
 
         // If we have extra, deposit back into `_bestVault` for `sender`
         // NOTE: Invariant is `withdrawn <= amount`
-        if (withdrawn > amount) {
+        if (withdrawn > amount && amount > _bestVault.pricePerShare()) {
```

Comment:
The problem isn't that it's less than what's required for 1 share, it's that it's less than what's required for `1 / 10 ** decimals()` part of a share.

Also, this would keep the dust "trapped" in whatever implements `BaseWrapper`

Reply:
I don't see how it will be kept as dust, https://github.com/steffenix/yearn-vaults/blob/2960055466d5ab68553cfa03f34f1674bb57fd98/contracts/BaseWrapper.sol#L263 won't be executed and the overall withdrawn will be sent back to the receiver

Reply:
> Yes indeed, should we expose `_shareValue`?

I think so

Reply:
> I don't see how it will be kept as dust, https://github.com/steffenix/yearn-vaults/blob/2960055466d5ab68553cfa03f34f1674bb57fd98/contracts/BaseWrapper.sol#L263 won't be executed and the overall withdrawn will be sent back to the receiver

Fair point.